src/platforms/f3/platform.c

```diff
@@ -112,9 +118,71 @@ bool platform_nrst_get_val(void)
 	return (gpio_get(NRST_PORT, NRST_PIN)) ? false : true;
 }
 
+
+static void adc_init(void)
+{
+
+	//ADC
+	rcc_periph_clock_enable(RCC_ADC12);
+	//ADC
+	gpio_mode_setup(VTREF_PORT, GPIO_MODE_ANALOG, GPIO_PUPD_NONE, VTREF_PIN);
+	//gpio_mode_setup(GPIOA, GPIO_MODE_ANALOG, GPIO_PUPD_NONE, GPIO1);
+	adc_power_off(ADC1);
+	adc_set_clk_prescale(ADC1, ADC_CCR_CKMODE_DIV2);
+	adc_set_single_conversion_mode(ADC1);
+	adc_disable_external_trigger_regular(ADC1);
+	adc_set_right_aligned(ADC1);
+	/* We want to read the temperature sensor, so we have to enable it. */
+	//adc_enable_temperature_sensor();
+	adc_set_sample_time_on_all_channels(ADC1, ADC_SMPR_SMP_601DOT5CYC);
+	uint8_t channel_array[] = { 1 }; /* ADC1_IN1 (PA0) */
```

Comment:
Rather than defining an array, please define just a channel value and take its address below as in:
```c
uint8_t channel = 1U;
adc_set_regular_sequence(ADC1, 1U, &channel);
```